### PR TITLE
Inkscape module

### PR DESCRIPTION
**Pull request by scribblemaniac**
This adds support for rendering vector files with Inkscape. This module has support for a bunch of file formats, width and height parameters, and automatic detection of linked files.

### CODE REVIEW

**Comment by morevnaproject** [comment]
@scribblemaniac Thank you for the contribution!

Regarding to "width" and "height" parameters I am still thinking about the right behavior.
When we deal with resizing, we have two possible situations:

1. We have files that must inherit the project resolution. 
Those are: Blender and Synfig files. We can define any  can define image size with arbitrary aspect and they will render the data without stretching (distortion).  So, those applications do *flexible resizing*.

2. Also, we can have asset files (like textures) - they have arbitrary resolution and shouldn't inherit project settings.
In other words, we have to get image dimensions from rendered file. The Krita and Inkscape files fall into this category. Pencil2D files also go here, since Pencil cannot do any resizing at all. ^__^

This is just thinking aloud, I am still not sure what to do here. Probably I am going to introduce a parameter that indicate which case the filetype belongs to. This parameter can be overriden for individual files using a .conf file. 

Not sure how to call this option... Something like "use_own_dimensions". ^__^

**Comment by scribblemaniac** [comment]
The new parameter thing sounds like a good idea. Another way of doing it would be to have default behaviors for each type (i.e. Synfig, and Blender use project width/height, and the rest use the resolution defined in the file). Then you define special keywords that can be used in the individual .conf files on a per parameter basis.

For example, Krita files would automatically use whatever size the image was created at. If it had an individual configuration file like this though:
```
WIDTH=inherit
HEIGHT=inherit
```
Then it would render at the size specified by the global project configuration file. The opposite could be done for a Blender file with a keyword like `source`.

Perhaps I'm overthinking this, but you could take it one step further by having the option to specify a path, relative to the main project directory, to inherit from. You may have for example, a shot that you want to render at a higher frame rate, but it contains animation assets. The assets may have a different resolution, but they all should share the same frame rate. It would be nice if you could have a configuration like this for an asset:
```
FRAMERATE=inherit 01/01Main.blend
WIDTH=800
HEIGHT=600
```
and then this asset would automatically match the frame rate that's found in 01/01Main.blend. This way, if you wanted to change the frame rate later on, you would only have to change one parameter to change all of the assets within that shot.

Of course this has some additional complexities with it. Depending on how .conf files are currently handled, this could be very easy or very difficult to implement. Also, you would need to set a max recursion depth, or something similar to handle scenarios such as the one where A inherits B and B inherits A. You would also have to add the configurations of inherited files as dependencies, so that when they change, the things that inherit its settings are updated as well.

Anyway, that's just my thinking aloud about you thinking aloud. ^_^

**Comment by morevnaproject** [comment]
@scribblemaniac I have implemented some advanced handling for width/height in this commit - https://github.com/morevnaproject/RenderChan/commit/c3e4141664a76425d6d681c795c6c09c37cd966e
It is assumed that module can retrieve the width/height of the file through analyze() function. At the moment only Krita module can do that. ^__^

**Comment by scribblemaniac** [comment]
It should be fairly trivial to get that working for gimp and inkscape. When I get a chance, I'll make another PR for them.
